http/http_client.hpp

```diff
@@ -524,13 +530,18 @@ class ConnectionInfo : public std::enable_shared_from_this<ConnectionInfo>
         {
             // Now let's try to resend the data
             state = ConnState::retry;
-            doResolve();
+            restartConnection();
```

Comment:
I think a comment here and a few other places would make this code more maintainable

Reply:
Ack

Reply:
I think a few more comments, a more descriptive comment on why would help..